subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy

```diff
@@ -501,42 +507,54 @@ class BuildScriptClasspathIntegrationSpec extends AbstractIntegrationSpec implem
         file("current/build.gradle").text = subprojectSource(currentTimestampJar)
 
         expect:
-        succeeds("printMessage")
+        succeeds("printMessage", "--info")
+
+        getArtifactTransformJarsByName("original/testClasses.jar").size() == 1
+        getArtifactTransformJarsByName("instrumented/testClasses.jar").size() == 1
+    }
+
+    void notInJar9Cache(String filename) {
+        inJar9Cache(filename, false)
+    }
 
-        getCachedTransformedJarsByName("testClasses.jar").size() == 1
+    TestFile inJar9Cache(String filename, boolean shouldBeFound = true) {
+        String fullpath = result.output.readLines().find { it.matches(">>>file:.*${filename}") }.replace(">>>", "")
+        assert fullpath.startsWith(jars9CacheDir.toURI().toString()) == shouldBeFound
+        return new TestFile(new File(URI.create(fullpath)))
     }
 
-    void notInJarCache(String filename) {
-        inJarCache(filename, false)
+    TestFile notInArtifactTransformCache(String filename) {
+        inArtifactTransformCache(filename, false)
     }
 
-    TestFile inJarCache(String filename, boolean shouldBeFound = true) {
+    TestFile inArtifactTransformCache(String filename, boolean shouldBeFound = true) {
         String fullpath = result.output.readLines().find { it.matches(">>>file:.*${filename}") }.replace(">>>", "")
-        assert fullpath.startsWith(cacheDir.toURI().toString()) == shouldBeFound
+        assert fullpath.startsWith(artifactTransformCacheDir.toURI().toString()) == shouldBeFound
```

Comment:
💭  This re-parsing of console outputs with internal knowledge of cache directory structures is such a hard to follow, brittle way to write these tests. And we have so many of them. We should think about how we could do this better, i.e. without relying on (this much) implementation detail.

Reply:
Let's add an issue for this.